WeakAuras/RegionTypes/Text.lua

```diff
@@ -169,76 +174,124 @@ local function modify(parent, region, data)
     end
   end
 
-  local UpdateText
-  if Private.ContainsAnyPlaceHolders(data.displayText) then
-    local getter = function(key, default)
-      local fullKey = "displayText_format_" .. key
-      if (data[fullKey] == nil) then
-        data[fullKey] = default
+  local containsCustomText = false
+  if Private.ContainsCustomPlaceHolder(data.displayText) then
+    containsCustomText = true
+  end
+
+  if not containsCustomText and type(data.conditions) == "table" then
+    for _, condition in ipairs(data.conditions) do
+      if type(condition.changes) == "table" then
+        for _, change in ipairs(condition.changes) do
+          if type(change.property) == "string"
+          and change.property == "displayText"
+          and type(change.value) == "string"
+          and Private.ContainsCustomPlaceHolder(change.value)
+          then
+            containsCustomText = true
+            break
+          end
+        end
       end
-      return data[fullKey]
     end
-    local formatters = Private.CreateFormatters(data.displayText, getter)
-    UpdateText = function()
-      local textStr = data.displayText;
-      textStr = Private.ReplacePlaceHolders(textStr, region, nil, false, formatters);
-      if (textStr == nil or textStr == "") then
-        textStr = " ";
+  end
+
+  if containsCustomText and data.customText and data.customText ~= "" then
+    data.customTextFunc = WeakAuras.LoadFunction("return "..data.customText)
+  else
+    data.customTextFunc = nil
+  end
+
+  function region:ConfigureTextUpdate()
+    local UpdateText
+    if self.displayText and Private.ContainsAnyPlaceHolders(self.displayText) then
+      local getter = function(key, default)
+        local fullKey = "displayText_format_" .. key
+        if (data[fullKey] == nil) then
+          data[fullKey] = default
+        end
+        return data[fullKey]
       end
+      local formatters = Private.CreateFormatters(self.displayText, getter)
```

Comment:
I would prefer it if the formatters were created outside ConfigureTextUpdate.

That is in the loop above, which checks for CustomPlaceHolders also create formatters for any potential text. (With slight adjustments to not create duplicated formatters.) Store these in the region and pass it into ReplacePlaceHolders. Having too many formatters in there is not a problem.